pkgs/native_assets_builder/lib/src/model/dry_run_result.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:native_assets_cli/native_assets_cli_internal.dart';
+
+import '../../native_assets_builder.dart';
+import 'build_result.dart';
+
+extension type DryRunResult(BuildResultImpl _buildResultImpl) {
```

Comment:
For symmetry, can we make a BuildStepResultImpl and also make BuildResult an extension type?
BuildResult shouldn't be more important than linkresult or dryrunresult.

Or make all of them interfaces and a single implementation.